### ISSUE TITLE
Move hyperkube binary and images out from main k/k repository

### ISSUE THREAD

**Issue by dims** [kubernetes/kubernetes#81760]
At one point hyperkube was going to take everything over but that day is gone:
https://github.com/kubernetes/kubernetes/issues/16508

We don't have any owners for hyperkube and we have pending issues like https://github.com/kubernetes/kubernetes/issues/80374 and https://github.com/kubernetes/kubernetes/issues/71719

hyperkube is a magnet for pulling all kinds of dependencies (especially cloud providers)

We asked this question once earlier in:
https://github.com/kubernetes/kubernetes/issues/69896

And it's easy to pull it out of k/k repository - POC is here : https://github.com/dims/hyperkube

floated the idea on mailing lists:
- https://groups.google.com/d/msg/kubernetes-sig-architecture/4uTHr74ucv0/KImea_0RFAAJ
- https://groups.google.com/d/msg/kubernetes-sig-release/p9WHaLjvFhM/lfKoNj5IDwAJ

found some folks who may want / be able to help: @feiskyer @adelina-t  @PatrickLang @markyjackson-taulia 

/area code-organization

**Comment by dims** [comment]
We will need the generated openapi check in if we want k/k to be vendorable, see https://github.com/kubernetes/kubernetes/pull/81239

**Comment by adelina-t** [comment]
Yeah, Hyperkube is of interest to our Windows testing effort, as we depend on it to deploy our test clusters with aks-engine. I can help with moving it and making the transition smooth.

**Comment by PatrickLang** [comment]
cc @ritazh @jackfrancis

**Comment by feiskyer** [comment]
@dims any updates on this?

**Comment by adelina-t** [comment]
I started some work on pulling it from the main k/k repo, I'm still slowly working on build scripts. Will update when I have a PR in a week or two ( depending on the volume of my other commitments ).

**Comment by dims** [comment]
thanks @adelina-t

**Comment by dims** [comment]
As mentioned earlier, i have https://github.com/dims/hyperkube and looks like @adelina-t will be surfacing another repository as well. We can pick either one ( preferably @adelina-t 's) and request a new repository under `kubernetes-sigs`.

We will need folks who want to own this code and take it forward. So if you are interested, please chime in on this issue.

cc @dghubble @PatrickLang @MikeSpreitzer @neolit123 @vllry @deads2k @justaugustus

**Comment by justaugustus** [comment]
@dims -- SIG Release is on the hook to own and I'll steward.

From the contributors you mentioned, I'll find some key approvers.

**Comment by neolit123** [comment]
i think we should create the k/k-sig hosted hyperkube repo and start moving bits to there.

**Comment by adelina-t** [comment]
You can have a look here at the initial move: https://github.com/adelina-t/hyperkube 

Hyperkube can be built, images can be built here.
Next steps: create dependancy pinning / vendor update scripts.

**Comment by dims** [comment]
@justaugustus @adelina-t thanks! https://github.com/adelina-t/hyperkube is a great way to get us started.

**Comment by dghubble** [comment]
@dims Kubernetes core still plans to package other container images like kube-apiserver, kube-scheduler, kube-controller-manager? The desire is specific to moving the hyperkube image management, not container images distributables right?

**Comment by dims** [comment]
@dghubble correct

**Comment by josiahbjorgaard** [comment]
Bug triage for release 1.17 here triaging bugs. Still good to go? Don't forget that code freeze is Nov. 18!

**Comment by uablrek** [comment]
@dims @adelina-t I am using hyperkube in test environment and would pretty much want to continue to do so. I can help with testing and contributions.

is it agreed that https://github.com/adelina-t/hyperkube is the repo to use?

**Comment by uablrek** [comment]
Oh, and what is the time frame? When will I not be able to do;
```
make WHAT=cmd/hyperkube
```
in the k8s dir?

**Comment by dims** [comment]
@uablrek when we have a new github repository where you can do the same thing.

@justaugustus do we want to do that this week?

**Comment by justaugustus** [comment]
@dims -- Yep, it's top of mind to get the ball rolling on the migration this week!

**Comment by justaugustus** [comment]
Migration issue opened in k/org: https://github.com/kubernetes/org/issues/1365

**Comment by liggitt** [comment]
If we want to provide the same functionality as hyperkube via another project, I think it would make more sense to do so with a [shim executable that delegates to actual binaries](https://github.com/liggitt/kubernetes/commit/1cf17e64d15f7df312a1020083bd2d979223375e), and to build an image that actually contained the official binaries. That could be done with a single go file, a Dockerfile, and a build script, with no go dependency on k8s.io/kubernetes.

**Comment by liggitt** [comment]
That lets the hyperkube image just be a packaging mechanism, and the hyperkube binary avoid dependency entanglement.

**Comment by dghubble** [comment]
In the immediate term, I noticed v1.17 beta images for hyperkube were never published. I was hoping to do my usual testing now that beta's are out, especially if its moving. Seems like its no longer in the release flow, but also doesn't have the new home?

**Comment by dims** [comment]
@dghubble we are re-working the hyperkube image like @liggitt mentioned, please follow along in https://github.com/kubernetes/kubernetes/pull/84662

**Comment by dims** [comment]
/close

TL;DR we are not moving the binary/image from k/k, we have an alternative container image based on a script now.
